plugins/tickets.py

```diff
@@ -0,0 +1,1821 @@
+import re
+import itertools
+import datetime
+import asyncio
+import logging
+import contextlib
+from functools import reduce
+from collections import namedtuple
+from enum import Enum, IntEnum
+from typing import List
```

Comment:
Should we do typing in the rest of the bot? I'm not very up to speed with how that works.